docs/learn/submit-data.md

```diff
@@ -58,6 +58,34 @@ The `GasCostPerBlobByte` and `GasCostPerTransactionByte` are parameters that
 could potentially be adjusted through the system's governance mechanisms. Hence,
 actual costs may vary depending on the current state of these parameters.
 
+#### Gas fee calculation
+
+The total fee for a transaction is calculated as the product of the gas limit
+for the transaction and the gas price set by the user:
+
+$\text{Total Fee} = \text{Gas Limit} \times \text{Gas Price}$
+
+The gas limit for a transaction is the maximum amount of gas that a user is
+willing to spend on a transaction. It is determined by both a static
+"fixed cost" and a dynamic cost based on the size of each blob involved
+in the transaction:
+
+$\text{Gas Limit} = \text{Fixed Cost} + \sum_{i=1}^{n} \text{Size of Blob}_i \times \text{Gas Cost Per Blob Byte}$
```

Comment:
we might want to clarify here that Size of blob is in shares, then we multiply by the share size multiplied by the gas cost per blob byte

Reply:
also I know we link the code in other places, but it would probably also be good to link it here as well since it might be best for folks to just read that small portion of code

https://github.com/celestiaorg/celestia-app/blob/32d247971386c1944d44bec1faeb000b1ff1dd51/x/blob/types/payforblob.go#L169-L181

Reply:
im starting to think that this is duplicate information of what is included in the page already, and may not be 100% necessary

Reply:
thanks for the feedback, added my best attempt at implementing it!